### ISSUE TITLE
Naming of mvt.ScaleGeo

### ISSUE THREAD

**Issue by ear7h** [go-spatial/tegola#630]
I also came across some code in `mvt/scale.go` that does not do strictly scaling (it also removes points, which is slightly misleading), and @ARolek  and I came up with the idea of renaming it `PrepareGeo` or something similar. This could do some sort of scaling, de-duplication, and standardize how we handle something like the last point of polygon.

Note that this issue might be moved to the `go-spatial/geom` repo

**Comment by ARolek** [comment]
I think the Scale operation should be separate from the "prepare" operation. Scaling is really a reprojection operation, where the coordinate system is changed but the underlying geometry does not gain or lose points. The "prepare" concept I was proposing would be for preparing geometries for MVT specific encoding, the most important being checking for the first and last coordinate being the same in a polygon, and if so, removing it.

The de-duplication of points in a multi point geometry (currently in the ScaleGeo) package, should be a separate routine called something along the lines of DeDupe and could live in the geom package. 

As I have been working my way through various parts of the encoding code, there are a few places where functions are handling too much, much more than their name implies. These ideas are around us moving to more focused functions so the implementer can compose various geoprocessing and encoding steps as they see fit.

**Comment by gdey** [comment]
This is function is doing a translate and a reprojection of coordinate systems. Keep that in mind. The translation is from Webmecator to tile coordinate. The other thing to remember that as we pull these pieces apart we are slowing downing the processing by OLog(n) each time. This is the reasoning behind the conflation of translation and removal of duplicate points, that are next to each other. I'm not saying we should not do it, just realize we may need to re-conflate later.

Prepare is not a good name for the function, `translate` is the more appropriate name.

**Comment by ARolek** [comment]
> The other thing to remember that as we pull these pieces apart we are slowing downing the processing by OLog(n) each time. This is the reasoning behind the conflation of translation and removal of duplicate points, that are next to each other.

Correct, but on the other hand, we're not giving the implementer the option to not invoke these operations since the concepts are conflated. In the case of a Multi Point, you may have two points on top of each other that are associated with different information but the de-duplication process removes them. This could be problematic for say heatmap implementations where point density is important.

> Prepare is not a good name for the function, translate is the more appropriate name.
For the other functionality validate or makeValidate maybe more appropriate.

There are two functions being discussed here: 
 - `Translate`/ `Reproject` / `Transform`: convert from geometry coordinates to tile coordinates.
 - `Prepeare` / `<alternative name>`: prepare the geometry specific for MVT encoding. The main situation that comes to mind is comparing the first and the last point of a polygon to make sure they're different. For example, GeoJSON wants the first and last point to be the same, where MVT does not. Having a `Prepare` function would apply this check to the incoming geometry and modify the geometry if necessary. The implementer may want to skip this step if they know their geometries already match the expected MVT structure thus avoiding the additional processing. 

I think this issue should be moved to the geom package once we move MVT encoding to it.

**Comment by gdey** [comment]
Yes, that makes sense to me.
